backend/sirius-web-diagrams-layout-api/src/main/java/org/eclipse/sirius/web/diagrams/layout/api/ILayoutService.java

```diff
@@ -21,4 +23,18 @@
  */
 public interface ILayoutService {
     Diagram layout(Diagram diagram);
+
+    /**
+     * A partial layout that layouts only impacted elements.
+     *
+     * @param diagram
+     *            The new diagram to layout.
+     * @param moveEvent
+     *            the {@link MoveEvent} that has trigger the new layout. Can be null if no event occurs.
+     * @param startingPosition
+     *            the {@link Position} to use to create new graphical elements (following a tool applied on the diagram
+     *            for instance.
```

Comment:
While not an issue per se, it feels odd since we have either a move event or a new element created event, not both.

Reply:
Can be done later